src/annotation/my_annotation/AdulthoodAnnotationProcessor.java

```diff
@@ -0,0 +1,33 @@
+package annotation.my_annotation;
+
+import java.lang.reflect.Field;
+import java.time.LocalDate;
+import java.time.Period;
+
+public class AdulthoodAnnotationProcessor {
```

Comment:
Utility classes should not have public constructors. They are collections of static members, so they are not intended to be instantiated.  Even abstract utility classes, which can be extended, should not have public constructors.
Java adds an implicit public constructor to every class which does not define at least one explicitly. Hence, at least one non-public constructor should be defined.
You can add a private constructor to hide the implicit public one.